lib/src/signed_video_h26x_common.c

```diff
@@ -884,12 +890,17 @@ update_linked_hash(signed_video_t *self, uint8_t *hash, size_t hash_size)
     if (hash_size != self->sign_data->hash_size) return SV_INVALID_PARAMETER;
   }
   gop_info_t *gop_info = self->gop_info;
-  uint8_t *stored_hash = &gop_info->linked_hashes[hash_size];
-  uint8_t *linked_hash = &gop_info->linked_hashes[0];
-  // Move the content of stored_hash to linked_hash
-  memmove(linked_hash, stored_hash, hash_size);
-  // Copy the new hash to stored_hash
-  memcpy(stored_hash, hash, hash_size);
+  uint8_t *new_hash = &gop_info->linked_hashes[hash_size * 2];  // Newest linked hash
+  uint8_t *middle_hash = &gop_info->linked_hashes[hash_size];  // Middle hash in the chain
+  uint8_t *oldest_hash = &gop_info->linked_hashes[0];  // Oldest hash in the chain
+
+  // Move middle_hash to oldest_hash
+  memmove(oldest_hash, middle_hash, hash_size);
+  // Move new_hash to middle_hash
+  memmove(middle_hash, new_hash, hash_size);
```

Comment:
Do you need to do this in 2 steps? I think it is possible to move 2 hashes in one go.